src/averageAges.js

```diff
@@ -54,8 +91,31 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const women = people.filter(human => human.sex === 'f');
+  const ages = [];
+
+  women.forEach(woman => {
+    const children = findChildren(people, woman, onlyWithSon);
+
+    children.forEach(child => {
+      ages[ages.length] = child.born - woman.born;
+    });
+  });
+
+  return ages.reduce((x, y) => {
+    return x + y;
+  }, 0) / ages.length;
+}
+
+function findChildren(people, mother, onlyWithSon) {
+  return people.filter(child => {
+    const isChild = child.mother === mother.name;
+    const maleChild = child.sex === 'm';
+
+    return onlyWithSon ? isChild && maleChild : isChild;
```

Comment:
You may simplify this a little
```suggestion
    return onlyWithSon
      ? child.mother === mother.name && child.sex === m
      : child.mother === mother.name
```